Decred Wallet/Features/App Launch/StartScreenViewController.swift

```diff
@@ -60,47 +60,56 @@ class StartScreenViewController: UIViewController {
     }
     
     func loadMainScreen() {
-        if !AppDelegate.walletLoader.isWalletCreated {
+        if AppDelegate.walletLoader.oneOrMoreWalletsExist {
+            self.checkStartupSecurityAndStartApp()
+        } else if DcrlibwalletWalletExistsAt(WalletLoader.appDataDir, BuildConfig.NetType) {
+            self.checkStartupSecurityAndLinkExistingWallet()
+        } else {
             self.displayWalletSetupScreen()
         }
-        else if StartupPinOrPassword.pinOrPasswordIsSet() {
-            self.promptForStartupPinOrPassword()
+    }
+    
+    func checkStartupSecurityAndStartApp() {
+        if StartupPinOrPassword.pinOrPasswordIsSet() {
+            self.promptForStartupPinOrPassword(completion: self.openWalletsAndStartApp)
         } else {
-            self.unlockWalletAndStartApp(pinOrPassword: "public") // unlock wallet using default public passphrase
+            self.openWalletsAndStartApp(startupPinOrPassword: "")
         }
     }
     
-    func displayWalletSetupScreen() {
-        let walletSetupController = WalletSetupViewController.instantiate().wrapInNavigationcontroller()
-        walletSetupController.isNavigationBarHidden = true
-        AppDelegate.shared.setAndDisplayRootViewController(walletSetupController)
+    func checkStartupSecurityAndLinkExistingWallet() {
+        if StartupPinOrPassword.pinOrPasswordIsSet() {
+            self.promptForStartupPinOrPassword(completion: AppDelegate.walletLoader.linkExistingWalletAndStartApp)
+        } else {
+            AppDelegate.walletLoader.linkExistingWalletAndStartApp(startupPinOrPassword: "")
```

Comment:
The default passphrase should be "public". the app crash when you try to unlock an existing wallet